packages/components/src/spectrum/View.tsx

```diff
@@ -43,34 +52,48 @@ export const View = forwardRef<DOMRefValue<HTMLElement>, ViewProps>(
       ...rest
     } = props;
 
-    // Using shorthand to define border styling. Removing a style property during rerender can lead to styling bugs
-    // ex. changing from borderXColor = blue to borderColor = blue will cause the left and right borders to disappear
-    const defaultBorderColor = colorValueStyle(borderColor) ?? 'transparent';
-    const defaultBorderXColor =
-      colorValueStyle(borderXColor) ?? defaultBorderColor;
-    const defaultBorderYColor =
-      colorValueStyle(borderYColor) ?? defaultBorderColor;
-    const topColor = colorValueStyle(borderTopColor) ?? defaultBorderYColor;
-    const bottomColor =
-      colorValueStyle(borderBottomColor) ?? defaultBorderYColor;
-    const leftColor = colorValueStyle(borderStartColor) ?? defaultBorderXColor;
-    const rightColor = colorValueStyle(borderEndColor) ?? defaultBorderXColor;
+    const style = useMemo(() => {
+      const borderStyle: CSSProperties = {};
+      if (borderColor !== undefined) {
+        borderStyle.borderColor = colorValueStyle(borderColor);
+      }
+      if (borderXColor !== undefined) {
+        borderStyle.borderLeftColor = colorValueStyle(borderXColor);
+        borderStyle.borderRightColor = colorValueStyle(borderXColor);
+      }
+      if (borderYColor !== undefined) {
+        borderStyle.borderTopColor = colorValueStyle(borderYColor);
+        borderStyle.borderBottomColor = colorValueStyle(borderYColor);
+      }
+      if (borderStartColor !== undefined) {
+        borderStyle.borderLeftColor = colorValueStyle(borderStartColor);
```

Comment:
Technically, this would be incorrect for RTL languages. We should be using `useLocale()` to get the direction of text, and setting the appropriate side depending on if it's RTL or LTR: https://github.com/adobe/react-spectrum/blob/bd458c1ed166a5ff1fd93f0c1a397e1540b3d880/packages/%40react-spectrum/utils/src/styleProps.ts#L259
So something like:
```
const { direction } = useLocale();

...
if (borderStartColor !== undefined) {
  borderStyle[direction === 'rtl' ? 'borderRightColor' : 'borderLeftColor'] = convertValueStyle(borderStartColor);
}
```

Alternatively you could find a way to use `useStyleProps` or `convertStyleProps` directly, and not have to add all this stuff... though that would mean passing in our own [viewStyleProps](https://github.com/adobe/react-spectrum/blob/bd458c1ed166a5ff1fd93f0c1a397e1540b3d880/packages/%40react-spectrum/utils/src/styleProps.ts#L66) with the `convertValueStyle` as the handler... probably simpler to just check rtl and do it here.

Reply:
I was able to use useStyleProps/convertStyleProps previously when I was doing background, so that is possible but it ultimately ended up simpler to not. 

As far as I am aware, there's no other components where this applies, so I would probably just do what you are suggesting instead.